values/time.go

```diff
@@ -108,6 +108,12 @@ func (t Time) Time() time.Time {
 // Mul will multiply the Duration by a scalar.
 // This multiplies each component of the vector.
 func (d Duration) Mul(scale int) Duration {
+	// If the duration is zero, do nothing.
+	// This prevents a zero value from becoming negative
+	// which is not possible.
+	if d.IsZero() {
```

Comment:
I'm curious, is there anywhere where if a negative zero managed to creep in, would it have adverse effects at runtime?

Reply:
I don't think so. I think it would just end up with the `String()` function printing `0ns` when the duration was negative zero instead of `-0ns`. It could also result in `IsNegative()` returning true when the value is zero which isn't supposed to be possible.